src/NLog/LogFactory.cs

```diff
@@ -1166,6 +1168,42 @@ public LogFactory LoadConfiguration(string configFile)
             return LoadConfiguration(configFile, optional: false);
         }
 
+        /// <summary>
+        /// Reload the current logging configuration and apply it.
+        /// </summary>
+        /// <returns>LogFactory instance for fluent interface</returns>
+        public LogFactory ReloadConfiguration()
+        {
+            InternalLogger.Info("Reloading configuration...");
+            var errorRaised = false;
+
+            if (_config != null)
```

Comment:
I'm doubting, shouldn't we check _configLoaded instead of _config?

Reply:
`_configLoaded` is a special state variable for LogFactory to ensure it doesn't try to auto-load Configuration-property after Close/Shutdown.